dial_queue.go

```diff
@@ -128,6 +135,9 @@ func (dq *dialQueue) control() {
 	)
 
 	defer func() {
+		stats.Record(dq.ctx, metrics.SuccessfulDialsPerQuery.M(atomic.LoadInt64(&dq.successes)))
```

Comment:
we probably should wait here for all the workers to stop before gathering these (with something like sync.waitgroup)